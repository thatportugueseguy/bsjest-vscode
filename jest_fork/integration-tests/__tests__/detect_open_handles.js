/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow
 */
'use strict';

const runJest = require('../runJest');

try {
  // $FlowFixMe: Node core
  require('async_hooks');
} catch (e) {
  if (e.code === 'MODULE_NOT_FOUND') {
    // eslint-disable-next-line jest/no-focused-tests
    fit('skip test for unsupported nodes', () => {
      console.warn('Skipping test for node ' + process.version);
    });
  } else {
    throw e;
  }
}

function getTextAfterTest(stderr) {
  return stderr.split('Ran all test suites.')[1].trim();
}

it('prints message about flag on slow tests', async () => {
  const {stderr} = await runJest.until(
    'detect-open-handles',
    [],
    'Jest did not exit one second after the test run has completed.',
  );
  const textAfterTest = getTextAfterTest(stderr);

  expect(textAfterTest).toMatchSnapshot();
});

it('prints message about flag on forceExit', async () => {
  const {stderr} = await runJest.until(
    'detect-open-handles',
    ['--forceExit'],
    'Force exiting Jest',
  );
  const textAfterTest = getTextAfterTest(stderr);

  expect(textAfterTest).toMatchSnapshot();
});

it('prints out info about open handlers', async () => {
  const {stderr} = await runJest.until(
    'detect-open-handles',
    ['--detectOpenHandles'],
    'Jest has detected',
  );
  const textAfterTest = getTextAfterTest(stderr);

  expect(textAfterTest).toMatchSnapshot();
});
