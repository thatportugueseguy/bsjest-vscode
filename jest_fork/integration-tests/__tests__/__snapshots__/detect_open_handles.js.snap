// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints message about flag on forceExit 1`] = `
"Force exiting Jest

Have you considered using \`--detectOpenHandles\` to detect async operations that kept running after all tests finished?"
`;

exports[`prints message about flag on slow tests 1`] = `
"Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with \`--detectOpenHandles\` to troubleshoot this issue."
`;

exports[`prints out info about open handlers 1`] = `
"Jest has detected the following 1 open handle potentially keeping Jest from exiting:

  ●  GETADDRINFOREQWRAP

      5 | const app = new http.Server();
      6 | 
    > 7 | app.listen({host: 'localhost', port: 0});
        |     ^
      8 | 
      
      at Object.<anonymous> (server.js:7:5)
      at Object.<anonymous> (__tests__/test.js:3:1)"
`;
